controller/src/main/java/cern/jarrace/controller/domain/Entrypoint.java

```diff
@@ -0,0 +1,54 @@
+package cern.jarrace.controller.domain;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * @author timartin
+ * Domain class that represents a registered {@link Entrypoint}
+ */
+
+@Entity
+@Table(name = "ENTRYPOINT_TABLE")
+public class Entrypoint {
+
+    /**
+     * Unique identifier for the {@link Entrypoint}
+     */
+    @Id
+    @Column(name = "AGENT_NAME", unique = true)
+    String agentName;
+
+    /**
+     * Port used by the {@link Entrypoint} to expose its services
+     */
+    @Column(name = "PATH", nullable = false)
+    String path;
+
+    // Needed by the JPA specification
+    public Entrypoint() {
+    }
```

Comment:
do you really need the declaration? it should be available as the default constructor implicitely